src/main/scala/reporting/Converter.scala

```
@@ -412,58 +538,53 @@ object Converter {
                   map += (field -> (recEntry, perm))
                 }
-              case _ =>
+              case FVFEntry(_, field, values, perms, sort) =>
+                val recEntry = mapLocalVar(
+                  Some(sort),
+                  values._1.getOrElse(termEval.asInstanceOf[VarEntry], values._2),
+                  heap,
+                  model,
+                  newEncountered,
+                  nullRefName
+                )
+                val perm = perms._1.getOrElse(termEval.asInstanceOf[VarEntry], perms._2)
+                map += (field -> (recEntry, perm))
+                  
+              case _ => 
             }
           }
           RefEntry(name, map)
         } else {
           RecursiveRefEntry(name)
         }
+
       case Some(sorts.Seq(elementSort)) =>
-        extractSequence(
-          model,
-          heap,
-          nullRefName,
-          name,
-          elementSort,
-          encountered
-        )
+        extractSequence(model, heap, nullRefName, name, elementSort, encountered)
+
+      case Some(sorts.Set(elementSort)) => extractSet(model, name, elementSort)
+      case Some(sorts.UserSort(_)) => termEval
       case _ => termEval
     }
   }
 
   def typeToSort(typ: ast.Type): Option[Sort] = {
-    //If this returns None, we can still try to evaluate the model entry
-    typ match {
-      case ast.Int  => Some(sorts.Int)
-      case ast.Bool => Some(sorts.Bool)
-      case ast.Perm => Some(sorts.Perm)
-      case ast.Ref  => Some(sorts.Ref)
-      case ast.SeqType(elementsType) =>
-        val elementSort = typeToSort(elementsType)
-        elementSort.map(x => sorts.Seq(x))
-      case _ => None
+    // if this returns None, we can still try to evaluate the model entry
+    try{
+      Some(symbolConverter.toSort(typ)) // simplify the logic with this
+    }catch{
+      case _: Throwable => None
```

Are we sure this isn't hiding errors we would like to see?